boa/src/syntax/parser/mod.rs

```diff
@@ -1,916 +1,1733 @@
+                _ => unimplemented!("{:?}", kind),
+            }
         }
-    }
 
-    /// Returns an error if the next symbol is not the punctuator `p`
-    #[inline(always)]
-    fn expect_punc(&mut self, p: Punctuator, routine: &'static str) -> Result<(), ParseError> {
-        self.expect(TokenData::Punctuator(p), routine)
+        if self
+            .next_if_skip_lineterminator(TokenKind::Punctuator(Punctuator::Spread))
+            .is_some()
+        {
+            let node = self.read_assignment_expression()?;
+            return Ok(PropertyDefinition::SpreadObject(node));
+        }
+
+        let prop_name = self
+            .next_skip_lineterminator()
+            .map(|tok| to_string(&tok.kind))
+            .ok_or(ParseError::AbruptEnd)?;
+
+        if self
+            .next_if_skip_lineterminator(TokenKind::Punctuator(Punctuator::Colon))
+            .is_some()
+        {
+            let val = self.read_assignment_expression()?;
+            return Ok(PropertyDefinition::Property(prop_name, val));
+        }
+
+        // TODO: Split into separate function: read_property_method_definition
+        if self
+            .next_if_skip_lineterminator(TokenKind::Punctuator(Punctuator::OpenParen))
+            .is_some()
+        {
+            let params = self.read_formal_parameters()?;
+
+            self.expect(
+                TokenKind::Punctuator(Punctuator::OpenBlock),
+                Some("property method definition"),
+            )?;
+
+            let body = self.read_block()?;
+
+            return Ok(PropertyDefinition::MethodDefinition(
+                MethodDefinitionKind::Ordinary,
+                prop_name,
+                Node::FunctionDecl(None, params, Box::new(body)),
+            ));
+        }
+
+        // TODO need to revisit this
+        // if let TokenKind::Identifier(name) = tok.kind {
+        //     if name == "get" || name == "set" {
+        //         let may_identifier = self.peek_skip_lineterminator();
+        //         if may_identifier.is_some()
+        //             && matches!(may_identifier.unwrap().kind, TokenKind::Identifier(_))
+        //         {
+        //             let f = self.read_function_expression()?;
+        //             let func_name = if let NodeBase::FunctionExpr(ref name, _, _) = f.base {
+        //                 name.clone().unwrap()
+        //             } else {
+        //                 panic!()
+        //             };
+        //             return Ok(PropertyDefinition::MethodDefinition(
+        //                 if name == "get" {
+        //                     MethodDefinitionKind::Get
+        //                 } else {
+        //                     MethodDefinitionKind::Set
+        //                 },
+        //                 func_name,
+        //                 f,
+        //             ));
+        //         }
+        //     }
+
+        //     return Ok(PropertyDefinition::IdentifierReference(name));
+        // }
```

Comment:
What should we do with this? I'm guessing it's related to `CoverInitializedName` parsing, right?

Reply:
Its not `CoverInitializedName` you're close, its `MethodDefinition` https://tc39.es/ecma262/#prod-MethodDefinition

We could park it (along with this code) an an issue for now? 
Or we tackle it as part of this PR

Reply:
We can park it, but I would remove the code, and add an explanation on what should be achieved here. Something like:
```TODO
	// TODO: `MethodDefinition` parsing
	// <https://tc39.es/ecma262/#prod-MethodDefinition>
```

Reply:
https://github.com/jasonwilliams/boa/issues/292 there you go